rdr_service/offline/study_nph_biobank_file_export.py

```diff
@@ -0,0 +1,254 @@
+# Script to create biobank file export and store the file in GCS Bucket
+from datetime import datetime
+from collections import defaultdict
+from functools import lru_cache
+from typing import List, Dict, Any, Iterable, Optional
+from zlib import crc32
+from json import dump
+from re import findall
+
+from sqlalchemy import and_
+from rdr_service import config
+from rdr_service.api_util import open_cloud_file
+from rdr_service.model.participant_summary import ParticipantSummary as RdrParticipantSummary
+from rdr_service.model.rex import ParticipantMapping as RexParticipantMapping
+from rdr_service.model.study_nph import (
+    StudyCategory,
+    Order,
+    OrderedSample,
+    SampleUpdate,
+    BiobankFileExport,
+    SampleExport
+)
+
+from rdr_service.dao.participant_summary_dao import ParticipantSummaryDao as RdrParticipantSummaryDao
+from rdr_service.dao.rex_dao import RexParticipantMappingDao
+from rdr_service.dao.study_nph_dao import (
+    NphStudyCategoryDao,
+    NphOrderDao,
+    NphOrderedSampleDao,
+    NphSampleUpdateDao,
+    NphBiobankFileExportDao,
+    NphSampleExportDao
+)
+
+
+FILE_BUFFER_SIZE_IN_BYTES = 1024 * 1024 # 1MB File Buffer
+
+
+def _get_latest_biobank_file_export() -> BiobankFileExport:
+    nph_biobank_file_export = NphBiobankFileExportDao()
+    with nph_biobank_file_export.session() as session:
+        return session.query(BiobankFileExport).order_by(
+            BiobankFileExport.created.desc()
+        ).first()
+
+
+def _get_sample_updates_since_last_file_export() -> Iterable[SampleUpdate]:
+    last_successful_biobank_file_export = _get_latest_biobank_file_export()
+    sample_update_dao = NphSampleUpdateDao()
+    with sample_update_dao.session() as session:
+        return session.query(SampleUpdate).filter(
+            SampleUpdate.created >= last_successful_biobank_file_export.created
```

Comment:
I may be missing this, but are these SampleUpdate's being created on the API code? Part of the tech design for the file creation specified that they would need to be created when we receive an order.

Reply:
@kenny-skaggs , currently @xtoller does not have access to `SampleUpdate` model in his branches. This PR needs to be merged and then dchan can make the change to always create `SampleUpdate` entries whenever a new `OrderedSample` is created / an existing one is updated.